addons/mail/models/mail_message.py

```diff
@@ -517,6 +489,7 @@ def message_format(self):
             message['is_discussion'] = message['subtype_id'] and subtypes_dict[message['subtype_id'][0]]['id'] == com_id
             message['is_notification'] = message['is_note'] and not message['model'] and not message['res_id']
             message['subtype_description'] = message['subtype_id'] and subtypes_dict[message['subtype_id'][0]]['description']
+            message['is_history'] = notifs.filtered(lambda x: (x.mail_message_id.id == message['id']) and (x.res_partner_id == self.env.user.partner_id)).is_read
```

Comment:
I just quiclky went to this line without checking the rest, but we will need at least to adapt this: 
message_format result can be sent to multiple user (notification mechanism). Thus, is_history is subjective, and won't make sense for some user. This is why, we have a needaction_partner_ids and not a is_needaction. So basicaly, we would need a history_partner_ids.
A quick draft of what I mean:

```
for notif in notifs:
....
if notif.is_read():
    notif_dict[mid]['history_partner_ids'].append(notif.res_partner_id.id)
else:
    notif_dict[mid]['needaction_partner_ids'].append(notif.res_partner_id.id)
....
message['needaction_partner_ids'] = notif_dict.get(message['id'], dict()).get('partner_id', [])
            message['is_note'] = message['subtype_id'] and subtypes_dict[message['subtype_id'][0]]['id'] == note_id
message['history_partner_ids'] = notif_dict.get(message['id'], dict()).get('partner_id', [])
            message['is_note'] = message['subtype_id'] and subtypes_dict[message['subtype_id'][0]]['id'] == note_id
```	
and adapt js of course

Thanks

Reply:
@Xavier-Do 
Changes Done.
Thanks :)